### PR TITLE
[index.d.ts] allow passing null to control props

### PR DESCRIPTION
**Pull request by franklixuefei**
**What**:
added `null` type to control props

**Why**:
`null` can be passed to control props, as per the actual implementation.

**How**:
change the `index.d.ts` file.

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export interface DownshiftState<Item> {
   highlightedIndex: number | null
   inputValue: string | null
   isOpen: boolean
-  selectedItem: Item
+  selectedItem: Item | null
```
Comment (franklixuefei) [inline typings/index.d.ts]: if `null` is passed in the props, then `selectedItem` in the state object will always be `null`.

File: typings/index.d.ts [hunk 2]
```diff
@@ -55,10 +55,10 @@ export interface DownshiftProps<Item> {
     changes: StateChangeOptions<Item>,
   ) => Partial<StateChangeOptions<Item>>
   itemCount?: number
-  highlightedIndex?: number
-  inputValue?: string
+  highlightedIndex?: number | null
+  inputValue?: string | null
   isOpen?: boolean
-  selectedItem?: Item
+  selectedItem?: Item | null
```
Comment (franklixuefei) [inline typings/index.d.ts]: `null`s should be allowed for control props.

**Comment by kentcdodds** [comment]
How would you like to be a TypeScript maintainer @franklixuefei? :)

**Comment by franklixuefei** [comment]
Sounds good! I can take that responsibility :-)

Frank

**Comment by kentcdodds** [comment]
Thanks so much for your help! I've added you as a collaborator on the project. Please make sure that you review the `other/MAINTAINING.md` and `CONTRIBUTING.md` files (specifically the bit about the commit messages and the git hooks) and familiarize yourself with the code of conduct (we're using [the contributor covenant](http://contributor-covenant.org/)). You might also want to [watch the repo](https://help.github.com/articles/watching-repositories/) to be notified when someone files an issue/PR. Please continue to make PRs as you feel the need (you can make your branches directly on the repo rather than your fork if you want). Thanks! And welcome to the team :)

**Comment by franklixuefei** [comment]
Thanks Kent!

I’ve already watched the repo. I’ll take some time in the afternoon to review the documentation and readmes.

Thanks!

Frank

**Pull request merged**